v2/internal/util/match/literal_matcher.go

```diff
@@ -3,26 +3,28 @@
  * Licensed under the MIT license.
  */
 
-package config
+package match
 
 import (
 	"strings"
+
+	"github.com/Azure/azure-service-operator/v2/internal/util/typo"
 )
 
 // literalMatcher is a StringMatcher that provides a case-insensitive match against a given string
 type literalMatcher struct {
 	literal string
 	matched bool
-	advisor *TypoAdvisor
+	advisor *typo.Advisor
 }
 
 var _ StringMatcher = &literalMatcher{}
 
-// newLiteralMatcher returns a new literalMatcher for the given string
-func newLiteralMatcher(literal string) *literalMatcher {
+// NewLiteralMatcher returns a new literalMatcher for the given string
+func NewLiteralMatcher(literal string) StringMatcher {
```

Comment:
Should be private. The public API for matchers is through `StringMatcher` and `NewStringMatcher`

```suggestion
// newLiteralMatcher returns a new literalMatcher for the given string
func newLiteralMatcher(literal string) StringMatcher {
```

Reply:
agreed